test/unit/ui.js

```diff
@@ -1,124 +1,215 @@
+    integrations = {
+      errorReporter: {notifyException: sinon.spy()},
+      tracker: 'test',
+    };
     config = {
       domain: 'buckets-o-stuff.myshopify.com',
       storefrontAccessToken: 123,
     };
     client = ShopifyBuy.buildClient(config);
+    client.config.domain = 'test-domain.myshopify.com';
     sinon.stub(client.shop, 'fetchInfo').resolves(shopFixture);
-    ui = new UI(client, {});
     script = document.createElement('script');
-    script.setAttribute('data-shopify-buy-ui', true);
+    script.setAttribute(DATA_ATTRIBUTE, true);
     document.body.appendChild(script);
   });
 
   afterEach(() => {
-    ui = null;
     document.body.removeChild(script);
   });
 
   describe('constructor', () => {
-    it('uses the client.config.domain instead of making a network call', () => {
-      const shopInfo = sinon.spy(client.shop.fetchInfo);
-      client.config.domain = 'test-domain.myshopify.com';
-      ui = new UI(client, {});
+    let resizeSpy;
+    let hostClickSpy;
+    let escSpy;
+    let postMessageSpy;
+    let appendStyleTagSpy;
+
+    beforeEach(() => {
+      resizeSpy = sinon.spy(UI.prototype, '_bindResize');
+      hostClickSpy = sinon.spy(UI.prototype, '_bindHostClick');
+      escSpy = sinon.spy(UI.prototype, '_bindEsc');
+      postMessageSpy = sinon.spy(UI.prototype, '_bindPostMessage');
+      appendStyleTagSpy = sinon.spy(UI.prototype, '_appendStyleTag');
+      ui = new UI(client, integrations, 'test');
+    });
+
+    afterEach(() => {
+      ui = null;
+      resizeSpy.restore();
+      hostClickSpy.restore();
+      escSpy.restore();
+      postMessageSpy.restore();
+      appendStyleTagSpy.restore();
+    });
+
+    it('sets client and domain from passed in client and client.config.domain', () => {
+      assert.equal(ui.client, client);
       assert.equal(ui.config.domain, client.config.domain);
-      assert(shopInfo.notCalled);
     });
-  });
 
-  describe('createCart', () => {
+    it('creates an empty array for iframeComponents', () => {
+      assert.deepEqual(ui.iframeComponents, []);
+    });
+
+    it('creates an empty components object holding product, cart, collection, productSet, modal, and toggle', () => {
+      const emptyComponentsObj = {
+        product: [],
+        cart: [],
+        collection: [],
+        productSet: [],
+        modal: [],
+        toggle: [],
+      };
+      assert.deepEqual(ui.components, emptyComponentsObj);
+    });
 
-    let initStub;
+    it('creates component types object for product, cart, collection, productSet, modal, toggle', () => {
+      const componentTypes = {
+        product: Product,
+        cart: Cart,
+        collection: ProductSet,
+        productSet: ProductSet,
+        toggle: CartToggle,
+      };
+      assert.deepEqual(ui.componentTypes, componentTypes);
+    });
 
-    beforeEach(() => {
-      initStub = sinon.stub(Cart.prototype, 'init').resolves();
+    it('sets an error reporter from integrations props', () => {
```

Comment:
The test below uses the variable name in camel case (`sets stylesOverrides from params`). Here we could be using errorReporter in camel case as well. Also, the word `props` could be replaced with `params` for consistency.